### DISCUSSION TITLE
Viewports in Paperspace exported to PDF, PNG or SVG not visible

### DISCUSSION THREAD

**Discussion by Jaime-Moniet** [mozman/ezdxf#554] [Q&A]
Hello, I am using the ezdxf example code from "viewports_in_paperspace.py"  as the basis in a little project I am working with, but have added the following code (within the make function of the example), to generate either png, pdf or svg files based on the ezdxf drawing / export addon but to export directly from the paperspace, where the viewports are:

`

        # Convert dxf to png
        fig = plt.figure()
        ctx = RenderContext(doc) 
        ax = fig.add_axes([0, 0, 1, 1])
        out = MatplotlibBackend(ax, params={"lineweight_scaling": 0.1,
        })
        Frontend(ctx, out).draw_layout(layout, finalize=True)
        fig.savefig("viewports_in_paperspace_R2007.png", dpi=300)
`
The Viewports are visible in AutoCAD, but the viewports in all the png, pdf or svg files are not visible, only the text added into the paperspace is visible.

In the command line, I will get multiple messages saying: "Warning: viewport with null view vector" (when  vp_layer is on, or "skipped entity VIEWPORT(#1D5). Reason: "invisible"" when the vp_layer is off.

I have tried explicitly using dxfattribs={"plot": 1} within the doc.new.layers() function but does not help.

Is there something I am missing here? Any tips would be appreciated.

**Comment by mbway** [comment]
viewports are not properly supported by the drawing addon. see https://github.com/mozman/ezdxf/discussions/535. However they should be drawn as rectangles.
The support is quite basic so there are lots of cases that might not be handled properly

**Reply by Jaime-Moniet** [reply]
Hi @mbway , thanks for your response. 

I am not getting even the rectangles of the viewports in the png file when I follow the example code described above. See image below:

![viewports_in_paperspace_R2007](https://user-images.githubusercontent.com/65253723/136668770-8c4e4c8d-ceb4-4e8b-9639-1021aa6f77ee.png)
o 

When I implement the drawing add on on a dxf file which had a viewport already created in AutoCAD, I can see a grey rectangle of that viewport in the png, PDF, svg files.

The reason I am interested in this drawing add on feature for viewports from the paperspace is because I was planning to use a drawing template/title block in paperspace. I guess a work around this issue is to import the drawing title block into modelspace just for the export into an svg, pdf or png file, which I have tested and know it works, but then have a separate routine to save the dxf with viewports in paperspace, which seems to work too.
